### PR TITLE
Adding jobs index on user_id

### PR DESCRIPTION
**Pull request by rikeoka**
* During the audit, the performance of deploys/search was extremely slow and often timed out with a 404 due to the volume of deploys/jobs we have.
* Adding a jobs table index on user_id (deployer) so the search queries would run faster.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: N/A

### Risks
- Level: Low

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -164,6 +164,7 @@
 
   add_index "jobs", ["project_id"], name: "index_jobs_on_project_id", using: :btree
   add_index "jobs", ["status"], name: "index_jobs_on_status", length: {"status"=>191}, using: :btree
+  add_index "jobs", ["user_id"], name: "index_jobs_on_user_id", using: :btree
```
Comment (rikeoka) [inline db/schema.rb]: When I run the migrations the schema.rb file now shows t.index, but it had a lot of changes, so manually updated the file. 

Should I push the new format schema.rb file?

Comment (grosser) [inline db/schema.rb]: It seems to always randomly change ... we can try normalizing it again but that should be a separate PR ... we had so many funky issues around that ... so I just stay with hand-picked updates ....

**Comment by grosser** [comment]
👍
